core/shared/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -101,24 +101,26 @@ package object internal extends InternalPlatform {
 
     val len = data.length
     if ((len & 0x01) != 0) None
-    val out = new Array[Byte](len >> 1)
-    var f: Int = -1
-    // two characters form the hex value.
-    try {
-      var i = 0
-      var j = 0
-      while (j < len) {
-        f = toDigit(data(j)) << 4
-        j += 1
-        f = f | toDigit(data(j))
-        j += 1
-        out(i) = (f & 0xff).toByte
+    else {
+      val out = new Array[Byte](len >> 1)
```

Comment:
Woah, it looks like previously there was a bug (missed explicit `return` in the if-statement). But won't we unintentionally get new behaviour?

Reply:
That is gratifying, as the goal is to inflict enough pleasure to compensate.